CRM/MembershipExtras/API/PaymentSchedule/MembershipType.php

```diff
@@ -13,20 +18,25 @@ class CRM_MembershipExtras_API_PaymentSchedule_MembershipType extends CRM_Member
    */
   public function __construct($params) {
     $this->params = $params;
-    $this->validateSchedule();
+    $this->membershipType = CRM_Member_BAO_MembershipType::findById($this->params['membership_type_id']);
```

Comment:
@omarabuhussein  I understand your point here, the reason we use call CRM BAO object instead of API because the Membership Type is using everywhere in the services from existing service like Duration calculator to a new service like Instalment schedule so If we use the API here we would need to convert get DAO object somewhere else, or we would need to write right all those class to handle Membership type as an array instead of DAO object.